include/nbl/ext/ImGui/ImGui.h

```diff
@@ -1,76 +1,133 @@
+		};
+
+		struct S_CREATION_PARAMETERS
+		{
+			struct S_RESOURCE_PARAMETERS
+			{
+				nbl::video::IGPUPipelineLayout* const pipelineLayout = nullptr;				//! optional, default layout used if not provided declaring required UI resources such as textures (required font atlas + optional user defined textures) & corresponding samplers
+				uint32_t count = 0x45u;														//! amount of total UI textures (and corresponding samplers)
+
+				struct S_BINDING_REQUEST_INFO												//! for a given pipeline layout we need to know what is intended for UI resources
+				{
+					uint32_t setIx,															//! descriptor set index for a resource	
+					bindingIx;																//! binding index for a given resource
+				};
+
+				const S_BINDING_REQUEST_INFO textures = { .setIx = 0u, .bindingIx = 0u },	//! optional, default texture binding request info used if not provided (set & binding index)
+				samplers = { .setIx = 0u, .bindingIx = 1u };								//! optional, default sampler binding request info used if not provided (set & binding index)		
+
+				using binding_flags_t = nbl::video::IGPUDescriptorSetLayout::SBinding::E_CREATE_FLAGS;
+				static constexpr auto TEXTURES_REQUIRED_CREATE_FLAGS = nbl::core::bitflag(binding_flags_t::ECF_UPDATE_AFTER_BIND_BIT) | binding_flags_t::ECF_PARTIALLY_BOUND_BIT | binding_flags_t::ECF_UPDATE_UNUSED_WHILE_PENDING_BIT; //! required flags
+				static constexpr auto SAMPLERS_REQUIRED_CREATE_FLAGS = nbl::core::bitflag(binding_flags_t::ECF_NONE); //! required flags
+				static constexpr auto RESOURCES_REQUIRED_STAGE_FLAGS = nbl::asset::IShader::E_SHADER_STAGE::ESS_FRAGMENT; //! required stage
+			};
+
+			nbl::asset::IAssetManager* const assetManager;									//! required		
+			nbl::video::IUtilities* const utilities;										//! required
+			nbl::video::IQueue* const transfer;												//! required
+			nbl::video::IGPURenderpass* const renderpass;									//! required
+			uint32_t subpassIx = 0u;														//! optional, default value used if not provided
+			S_RESOURCE_PARAMETERS resources;												//! optional, default parameters used if not provided
+			nbl::video::IGPUPipelineCache* const pipelineCache = nullptr;					//! optional, no cache used if not provided
+			typename MDI::COMPOSE_T* const streamingBuffer = nullptr;						//! optional, default MDI buffer allocated if not provided	
+		};
+
+		//! parameters which may change every frame, used with the .update call to interact with ImGuiIO; we require a very *required* minimum - if you need to cover more IO options simply get the IO with ImGui::GetIO() to customize them (they all have default values you can change before calling the .update)
+		struct S_UPDATE_PARAMETERS
+		{
+			//! what we pass to ImGuiIO::AddMousePosEvent 
+			nbl::hlsl::float32_t2 mousePosition,
+
+			//! main display size in pixels
+			displaySize;
```

Comment:
don't do such weird tricks like a `,` on a previous line, makes code unreadable and unextendable, type out the type

Reply:
your code